plugin/pkg/shoot/validator/admission.go

```diff
@@ -508,11 +508,18 @@ func validateKubernetesVersionConstraints(constraints []core.ExpirableVersion, s
 	validValues := []string{}
 	var latestVersion *semver.Version
 	for _, versionConstraint := range constraints {
-		if versionConstraint.ExpirationDate != nil && versionConstraint.ExpirationDate.Time.Before(time.Now()) {
+		if versionConstraint.ExpirationDate != nil && versionConstraint.ExpirationDate.Time.UTC().Before(time.Now().UTC()) {
 			continue
 		}
 
-		validValues = append(validValues, versionConstraint.Version)
+		if !getLatestPatchVersion {
+			validValues = append(validValues, versionConstraint.Version)
+		} else {
+			// ignore preview versions for defaulting
+			if versionConstraint.Classification != nil && *versionConstraint.Classification == core.ClassificationPreview {
```

Comment:
Can you explain why you want to ignore preview versions? Let's suppose `1.18.0` is a `preview` version and a shoot is created with `1.18`. Why wouldn't you want to default it to `1.18.0`? Instead the user only gets non 1.18 versions in the error message.

Reply:
Because that is the nature of preview versions (also agreed upon with @vlerenc ) - they should be opt-in. They should not be defaulted to. 
We should not default the stakeholders cluster to a version that is not thoroughly tested.
Gardener should default to stable versions and tested versions - we call them "supported".

This defaulting feature was to my knowledge created from the need of staekholders to create up to date clusters from their CI pipelines to test their workload on. Providing them with preview clusters is IMHO not helpful in this context.

Reply:
Okay, let's say it this way: `1.18.0` is in `preview` and it's also the latest release you offer. `1.17.3` is another version offered.

As far as I see the logic you'll get the following results:
1. Create a shoot with `1.18` --> Error message that `1.17.3` is a valid version.
2. Create a shoot with `1.19.1` --> Error message that `1.17.3` and `1.18.0` are valid versions.

Can you double check that pls? And if that's the case, it's a little strange, isn't it?

Reply:
At the moment it is intended to be exactly how you described in your example. 
I do not think that this is a strange behaviour.

1. 1.18 ask Gardener to default the version for the Shoot. Gardener does not default to preview versions -> 1.17.3 is the latest version it can default to.
2. 1.19.1 does not use defaulting, so preview versions can be selected on an opt-in basis. Thats why giving the valid values of 1.18.0 and 1.17.3 is correct in this context.

@rfranzke maybe you have an opinion here as well

Reply:
> 1.18 ask Gardener to default the version for the Shoot.

I'm not sure if users really use Gardener to indirectly ask what patch version it's capable of defaulting to. Imo, this is a convenience for the user and if we can't default a version why not showing all the versions the user can pick for his cluster (like done in the 2nd scenario). Easier for the user to understand and easier to maintain.

But sure, let's wait for other opinions or have an offline discussion :)

Reply:
I tend to agree with @timuthy - we should a) make the error messages clear and helpful, and b) prevent to implement any hard-to-maintain special cases. So, it's fine to not default to 1.18.0 if it's the only version and it's not yet supported/still in preview, but then let's show this in the error message ("supported versions are 1.18.0 (preview, must be selected explicitly) and 1.17.3").

Reply:
Ok, I think that makes sense, this way the user knows what is going on. I ll adjust it.
Thanks for fir reviewing so thoroughly!

Reply:
Added a separate commit containing the improved error message. PTAL